src/loaders/video_loader.py

```diff
@@ -29,6 +32,14 @@ def load(self):
                 _, frame = video.read()
                 continue
 
+            if self.threshold > 0:
```

Comment:
I am assuming this logic handles the frames which will be passed to your UDF. 

If this logic is an optimization specific to your UDF, then I feel this code doesn't belong here.